### PR TITLE
Replace react-script with vite

### PR DESCRIPTION
**Pull request by geichelberger**
Vite was chosen because of https://github.com/opencast/opencast-editor/pull/1265

> Vite was chosen because:
>
>    Straightforward migration
>    Relatively similar set of opinions as CRA
>    Optimized for single page applications like ours
>    Popular

I didn't reimplement the dev mode with the mocks.

### CODE REVIEW

**Comment by geichelberger** [comment]
Was the code necessary to deploy it like that?

File: package.json [hunk 1]
```diff
@@ -9,10 +9,10 @@
     "@mui/styles": "^5.15.18",
     "@mui/x-date-pickers": "^6.19.0",
     "@reduxjs/toolkit": "^1.9.7",
-    "@testing-library/jest-dom": "^5.17.0",
-    "@testing-library/react": "^12.1.5",
     "@types/dompurify": "^3.0.5",
+    "@types/http-proxy": "^1.17.14",
```
Comment (Arnei) [inline package.json:13]: This seems to be going unused.

File: package.json [hunk 2]
```diff
@@ -74,10 +72,21 @@
...
     "@babel/preset-env": "^7.23.8",
     "@babel/preset-react": "^7.24.1",
+    "@redux-devtools/extension": "^3.3.0",
     "@types/lodash": "^4.14.202",
+    "@types/node": "^20.12.12",
     "@types/uuid": "^9.0.8",
+    "@vitejs/plugin-react": "^4.2.1",
+    "eslint": "^8.56.0",
+    "eslint-plugin-react": "^7.33.2",
+    "eslint-plugin-react-hooks": "^4.6.0",
```
Comment (Arnei) [inline package.json:82]: Explicitly importing these seems to be unnecessary?

File: package.json [hunk 3]
```diff
@@ -74,10 +72,21 @@
     "@babel/plugin-syntax-optional-chaining": "^7.8.3",
     "@babel/preset-env": "^7.23.8",
     "@babel/preset-react": "^7.24.1",
```
Comment (Arnei) [inline package.json:74]: Gettring rid of CRA should also mean that we can get rid of babel!

**Review by Arnei — CHANGES_REQUESTED** [review]
Good riddance CRA, hello Vite!

Works and looks good to me, the comments below are just nitpicks, I am more than willing to approve this. Although we should still discuss if we want to get rid of the mock data. (Personally I never use it so I wouldn't mind if it was gone).

**Review by Arnei — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -26,29 +26,17 @@ This runs a development server at `http://localhost:3000`, serving a development
...
-This assumes you have an Opencast instance running at `http://localhost:8080`
-to which the development server will then proxy all the backend request,
+This assumes you have an internet connection to which the development server will then proxy all the backend request,
 authenticating them as user `admin` with password `opencast`.
 
 If you want to work with a different Opencast and/or user, you can change the command thusly:
 
 ```sh
-PROXY_TARGET=https://develop.opencast.org npm start
+PROXY_TARGET=http://localhost:8080 npm start
```
Comment (ziegenberg) [inline README.md:36]: Why not leave the example with `https://develop.opencast.org` in here? It would make it easier for the testers to know what to type.

Comment (geichelberger) [reply README.md:36]: This is the example for a different target, the default is `https://develop.opencast.org`

Comment (ziegenberg) [reply README.md:36]: Ah, I overlooked the config `vite.config.ts`

**Comment by geichelberger** [comment]
As the test deployments are only static files, removing the proxy should not cause any trouble.

**Review by lkiesow — APPROVED** [review]
Misunderstood that:
- Mock data is gone (that just needs a slight text update in #470)
- Proxy still exists
- Default target now is https://develop.opencast.org which I like since it should work for most people with the default config

File: README.md [hunk 5]
```diff
@@ -26,29 +26,17 @@ This runs a development server at `http://localhost:3000`, serving a development
...
-with real data, or need the ability to change it, you can rely on the
-proxy functionality of said development server, instead of running the static file server. Run:
-
-```sh
-PROXY=1 npm start
-```
-
-This assumes you have an Opencast instance running at `http://localhost:8080`
-to which the development server will then proxy all the backend request,
+This assumes you have an internet connection to which the development server will then proxy all the backend request,
```
Comment (lkiesow) [inline README.md:30]: Can we explicitly state that the default proxy target is https://develop.opencast.org? That “the internet” is proceed is rather vague :grin:

**Comment by lkiesow** [comment]
Since #470 got merged, can you update the test comment:

https://github.com/opencast/opencast-admin-interface/blob/ddfd555a61efcbb11f0376b07724975327657446/.github/workflows/deploy-container-image.yaml#L53-L61

This should now probably be something like:

     Run test server using develop.opencast.org as backend:
     ``` 
     podman run --rm -it -p 127.0.0.1:3000:3000 ${{ steps.meta.outputs.tags }} 
     ``` 
  
     Specify a different backend like stable.opencast.org:
     ``` 
     podman run --rm -it -p 127.0.0.1:3000:3000 -e PROXY_TARGET=https://stable.opencast.org ${{ steps.meta.outputs.tags }} 
     ```

**Comment by lkiesow** [comment]
Sorry for finding another issue :see_no_evil: 

This causes the container to fail since they do not contain a browser:
```
❯ podman run --rm -it -p 127.0.0.1:3000:3000 opencast-admin-interface:6dfe2dc                                           

> admin-ui-picard@0.1.0 start
> vite

The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.

  VITE v5.2.11  ready in 207 ms

  ➜  Local:   http://localhost:3000/
  ➜  Network: use --host to expose
node:events:497
      throw er; // Unhandled 'error' event
      ^

Error: spawn xdg-open ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:292:12)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn xdg-open',
  path: 'xdg-open',
  spawnargs: [ 'http://localhost:3000/' ]
}

Node.js v20.13.1
```

The vite server also listens to localhost only which does not work with the containers.

There may be a better way, but this will fix the issues:

```diff
diff --git a/Dockerfile b/Dockerfile
index 2782425b1f..1890183c3c 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -7,4 +7,5 @@ WORKDIR /admin-interface
 RUN npm ci
 
 ENV CI true
+ENV BROWSER none
 CMD [ "npm", "start" ]
diff --git a/package.json b/package.json
index 4a14ced112..c0bd48faa9 100644
--- a/package.json
+++ b/package.json
@@ -45,7 +45,7 @@
     "yup": "^1.4.0"
   },
   "scripts": {
-    "start": "vite",
+    "start": "vite --host 0.0.0.0",
     "build": "tsc && vite build",
     "serve": "vite preview",
     "test": "vitest"
```

**Comment by geichelberger** [comment]
I did not alter the package.json because I would not expect vite to listen on all addresses in a dev environment. Instead, I added the host option to the `CMD` instruction.

**Comment by lkiesow** [comment]
Great. Tried that but without the `--`. Didn't know that I need to separate that. Thanks. Merging…

**Pull request merged**